pandas/core/common.py

```diff
@@ -2,23 +2,66 @@
 Misc tools for implementing data structures
 """
 
-import re
-import collections
-import numbers
+import sys
+import warnings
 from datetime import datetime, timedelta
 from functools import partial
 
 import numpy as np
-import pandas as pd
-import pandas.algos as algos
 import pandas.lib as lib
 import pandas.tslib as tslib
 from pandas import compat
-from pandas.compat import (long, zip, map, string_types,
-                           iteritems)
-from pandas.types import api as gt
-from pandas.types.api import *  # noqa
+from pandas.compat import long, zip, iteritems
 from pandas.core.config import get_option
+from pandas.types.generic import ABCSeries
+from pandas.types.common import _NS_DTYPE, is_integer
+from pandas.types.inference import _iterable_not_string
+from pandas.types.missing import isnull
+from pandas.api import types
+from pandas.types import common
+
+# back-compat of public API
+# deprecate these functions
+m = sys.modules['pandas.core.common']
+for t in [t for t in dir(types) if not t.startswith('_')]:
+
+    def outer(t=t):
+
+        def wrapper(*args, **kwargs):
+            warnings.warn("pandas.core.common.{t} is deprecated. "
+                          "import from the public API: "
+                          "pandas.api.types.{t} instead".format(t=t),
+                          FutureWarning, stacklevel=2)
```

Comment:
@jreback I think in this case  case we should deviate from our normal rule of using FutureWarning for deprecations, and actually use a DeprecationWarning. 
Typically, if people have used this, it will rather be in a package than in interactive explorative coding. And I think it is not needed that users of those packages see this warning (as they can't change it themselves)